content/en/docs/concepts/containers/images.md

```diff
@@ -55,9 +55,9 @@ pulling an image if it already exists. If you would like to always force a pull,
 you can do one of the following:
 
 - set the `imagePullPolicy` of the container to `Always`.
-- omit the `imagePullPolicy` and use `:latest` as the tag for the image to use.
-- omit the `imagePullPolicy` and the tag for the image to use.
 - enable the [AlwaysPullImages](/docs/reference/access-authn-authz/admission-controllers/#alwayspullimages) admission controller.
+- omit the `imagePullPolicy` and either use `:latest` as the tag for the image, or omit the tag entirely.
```

Comment:
Is the earlier wording
> The default pull policy is `IfNotPresent`

accurate?

Reply:
I'm not sure I understand? All four of the bullet points in this list are introduced by "If you would always like to force a pull," so none of these are about `IfNotPresent`.

Reply:
I wonder if this advice is right _in context_. If I omit the `imagePullPolicy` and then Kubernetes sets it to its default value (the page says the default policy is `IfNotPresent`), how does that work? It's going to confuse readers. I want to minimize the chance of a reader ending up more puzzled than before they read the page.

Reply:
Would it help to separate the statement about the default pull policy from the list of policy/tag settings?
Is this true:

If you omit or do not set the `imagePullPolicy` field in your configuration, the image pull policy defaults to `IfNotPresent`. When `IfNotPresent` is set, the  
{{< glossary_tooltip text="kubelet" term_id="kubelet" >}} skips pulling an image if the image already exists.

If you would like to always force a pull, you can do one of the following:

Reply:
According to the API spec, `imagePullPolicy` defaults to `"Always"` if the `":latest"` tag is specified, or `"IfNotPresent"` otherwise. Isn't this clear enough?

Reply:
@tengqm - this doesn't actually work as you might expect, as I discovered (and wrote about) here: https://itnext.io/defaults-are-hard-kubernetes-deployment-edition-3b11095792f2

Reply:
ah, got it. Thank you for the clarification, @adrianludwin. It smells to me like a bug in the Container update code path. When the new image now contains `":latest"`, the `imagePullPolicy` should switch to `"Always"` immediately. This is something we need to bring @kubernetes/sig-api-machinery-api-reviews  @kubernetes/sig-apps-api-reviews teams in.

Reply:
A quick dive into the source ... I'm wondering if something can be done at [here](https://github.com/kubernetes/kubernetes/blob/master/pkg/registry/core/pod/strategy.go#L99-L100), the Pod strategy `canonicalize()` function. That function supposedly run after validation for create/update. @liggitt

Reply:
> When the new image now contains ":latest", the imagePullPolicy should switch to "Always" immediately.

The issue is on an update which only changes the image, while leaving the `imagePullPolicy` set to the existing value of `IfNotPresent`. Defaulting only applies when the `imagePullPolicy` field is absent.

Reply:
So another possibility is to validate if `imagePullPolicy` is properly changed if the new container image has `:latest` tag? https://github.com/kubernetes/kubernetes/blob/master/pkg/apis/core/validation/validation.go#L3832

Reply:
`image: "foo:latest", imagePullPolicy: "IfNotPresent"` could be a valid configuration, so we would change validation to forbid it, but we could add a non-fatal warning if someone changes an image from a non-`latest` tag to a `latest` tag and does not also modify `IfNotPresent` to `Always`

Reply:
I see. We are not supposed to forbid such a valid configuration...

Reply:
> Defaulting only applies when the `imagePullPolicy` field is absent.

Let's make that explicit in the docs. There are some subtle things to take in here.

I also want the reader to appreciate that there _isn't_ an “unset” value for `imagePullPolicy` in a container specification, at least not once that spec is persisted into a Kubernetes cluster via the API.
Nor is there any “please guess for me, kubelet” value.

Reply:
Let's continue this discussion on #26661.